optimistix/_solver/bisection.py

```diff
@@ -74,11 +136,26 @@ def init(
             lower_neg = lower_val < 0
             upper_neg = upper_val < 0
             flip = lower_val > upper_val
-            flip = eqx.error_if(
-                flip,
-                lower_neg == upper_neg,
-                msg="The root is not contained in [lower, upper]",
-            )
+            root_not_contained = lower_neg == upper_neg
+            if self.expand_if_necessary:
+                initial_interval = _ExpansionCarry(lower, upper)
+                need_positive = lower_neg
+                final_interval = jax.lax.cond(root_not_contained,
+                                              ft.partial(_expand_interval_repeatedly,
+                                                         fn=fn, args=args),
```

Comment:
Optionally, by the way, I think we can improve compile time here rather a lot: I suspect it should be possible to fold the two `fn` evaluations above into this loop. We just always run at least one (or two?) iterations, equivalent to the two evaluations we're currently doing.

This would both remove two callsites of `fn` and remove the surrounding `cond`.

I can see that might be a fiddlier change, though, so I'm not asking for that as a requirement to get this merged.

Reply:
I was able to remove the `cond` as requested (it's actually how I originally had it), but I can't see how to fold the functions in.